web/src/narrow.js

```diff
@@ -504,9 +513,12 @@ export function activate(raw_operators, opts) {
 
             message_fetch.load_messages_for_narrow({
                 anchor,
-                cont() {
+                anchor_date,
+                cont(data, opts) {
                     if (!select_immediately) {
                         update_selection({
+                            anchor_type: opts.anchor,
+                            anchor: data.anchor,
```

Comment:
I have to add these changes to update the focus view after fetching the response from the server (along with 
 [these changes](https://github.com/zulip/zulip/pull/25677/files#diff-6c6eeaa26f7145a3d41a485431ee2442ca53a7aded6a4fdbc265e9575326834dR796-R800)) because:
 - as suggested in the [chat](https://chat.zulip.org/#narrow/stream/14-GSoC/topic/Akshat.20-.20checkins/near/1590094) to use `then_select_id` logic to update the focus - the current code path doesn't work that way - it doesn't use `then_select_id` logic rather it calls this callback function `cont` in the very end to update the focus
 - the `near:` filter works differently and we can't leverage the same logic for this `date:` filter because `near` has a message ID specified which we may either find or not. This same message ID is set to the `id_info` object which is passed to this `cont` function (and hence the `near:` filter does not direct depend on the fetched server anchor value to update the focus). The correct value if found by the `closest_id` function in `MessageList` (all this happens inside the callback function `cont`).